sdk/src/azure/iot/az_iot_provisioning_client.c

```diff
@@ -558,3 +561,42 @@ AZ_NODISCARD az_result az_iot_provisioning_client_parse_received_topic_and_paylo
 
   return AZ_OK;
 }
+
+AZ_NODISCARD az_result az_iot_provisioning_client_get_request_payload(
+    az_iot_provisioning_client const* client,
+    az_span custom_payload_property,
+    az_iot_provisioning_client_payload_options const* reserved,
+    uint8_t* mqtt_payload,
+    size_t mqtt_payload_size,
+    size_t* out_mqtt_payload_length)
+{
+  (void)reserved;
+
+  _az_PRECONDITION_NOT_NULL(client);
+  _az_PRECONDITION_IS_NULL(reserved);
+  _az_PRECONDITION_NOT_NULL(mqtt_payload);
+  _az_PRECONDITION(mqtt_payload_size > 0);
+
+  az_json_writer json_writer;
+  az_span payload_buffer = az_span_create((uint8_t*)mqtt_payload, (int32_t)mqtt_payload_size);
+
+  _az_RETURN_IF_FAILED(az_json_writer_init(&json_writer, payload_buffer, NULL));
+  _az_RETURN_IF_FAILED(az_json_writer_append_begin_object(&json_writer));
+  _az_RETURN_IF_FAILED(az_json_writer_append_property_name(&json_writer, prov_registration_id_label));
+  _az_RETURN_IF_FAILED(az_json_writer_append_string(&json_writer, client->_internal.registration_id));
+
+  if  (az_span_size(custom_payload_property) > 0)
```

Comment:
Is writing an empty payload not valid?

For example, like this:
{
...
"payload" : ""
...
}